### PR TITLE
Add HTTP endpoint to get salon names

### PR DESCRIPTION
**Pull request by kdious**
/deploy/get_salon_info returns a JSON body
response that contains info for each salon
indexed by salon name.

### CODE REVIEW

**Comment by kdious** [comment]
👓 @spladug

**Comment by kdious** [comment]
Sanity checked this one as well.  Getting a 200 response.  I don't have any salons set up on my local dev environment but at least this returns a 200 response.  Should work in production I think.

```
Python 2.7.12 (default, Aug 22 2019, 16:36:40) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> import hmac
>>> import hashlib
>>> import urllib
>>> import urlparse
>>> 
>>> secret = 'asupersecrettoken'
>>> data = {'reason': 'Test', 'salon-name': 'test-salon'}
>>> body = urllib.urlencode(data)
>>> body = ''
>>> hash = hmac.new(secret, body, hashlib.sha1)
>>> headers = {"Content-Type": "application/x-www-form-urlencoded", "X-Hub-Signature": "sha1=" + hash.hexdigest()}
>>> 
>>> url = 'http://0.0.0.0:8080/harold/deploy/get_salon_info'
>>> requests.post(url, headers=headers, data=body)
<Response [200]>

```

**Comment by kdious** [comment]
👓 @andytuba

File: harold/plugins/deploy.py [hunk 1]
```diff
@@ -47,6 +47,36 @@ def __init__(self, http, monitor):
...
+
+    def _handle_request(self, request):
+        salons = yield self.salons.all()
+
+        salons_by_name = {}
+        for salon in salons:
+            salon_info = {
+                'name': salon.name,
+                'channel': salon.channel,
+                'allow_deploys': salon.allow_deploys,
```
Comment (andytuba) [inline harold/plugins/deploy.py]: Per slack conversation, let's only expose the data needed to support the current use case.

**Comment by kdious** [comment]
💇

File: harold/plugins/deploy.py (hunk 1 above)
Comment (kdious) [reply harold/plugins/deploy.py]: Done.

**Pull request merged**